api/cache.go

```diff
@@ -50,12 +50,7 @@ func (utc *userTierCache) Get(sub string) (int, bool) {
 func (utc *userTierCache) Set(u *database.User) {
 	var ce userTierCacheEntry
 	now := time.Now().UTC()
-	if u.SubscribedUntil.Before(now) {
-		ce = userTierCacheEntry{
-			Tier:      database.TierFree,
-			ExpiresAt: now.Add(userTierCacheTTL),
-		}
-	} else if u.QuotaExceeded {
+	if u.QuotaExceeded {
```

Comment:
are you sure we are correctly handling the webhooks? that happens in `processStripeSub`, I might be reading that wrong but I'm not 100% convinced that we appropriately update the user's tier based on the subscription start and end period. Do you mind verifying that? If it's fine then I'll approve but this might introduce more issues than it's causing and it's a dangerous PR because we're relying on something that's not in the diff

Reply:
I believe it works as expected. It hasn't been touched for months and I haven't heard of any indications that it's not working as expected. Also, if that handling is buggy, then the subscription expiry date in the user's record will also be off because it's set by the same function.